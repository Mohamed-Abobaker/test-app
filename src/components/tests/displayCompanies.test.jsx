import React from 'react';
import { shallow } from 'enzyme';
import DisplayCompanies from '../displayCompanies';

describe('DisplayCompanies.jsx tests', () => {
  it('should render the correct component', () => {
    const wrapper = shallow(<DisplayCompanies company={{}} index={0} updateCompanyIndex={() => { }} updateCompanies={() => { }} companies={[]} />)

    expect(wrapper.debug()).toMatchSnapshot();
  });
  it('should call correct function when update button is clicked', () => {
    const mockFunc = jest.fn();

    const wrapper = shallow(<DisplayCompanies company={{}} index={0} updateCompanyIndex={mockFunc} updateCompanies={() => { }} companies={[]} />)
    const buttons = wrapper.find('button')
    buttons.at(0).props().onClick();

    expect(mockFunc).toHaveBeenCalled();
  });
  it('should call the correct function when the remove button is clicked', () => {
    const mockFunc = jest.fn();

    const wrapper = shallow(<DisplayCompanies company={{}} index={0} updateCompanyIndex={() => { }} updateCompanies={mockFunc} companies={[{}]} />)
    const buttons = wrapper.find('button')
    buttons.at(1).props().onClick();

    expect(mockFunc).toHaveBeenCalled();
  });
});