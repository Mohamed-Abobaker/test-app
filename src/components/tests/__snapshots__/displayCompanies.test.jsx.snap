// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayCompanies.jsx tests should render the correct component 1`] = `
"<div>
  <p />
  <button onClick={[Function: onClick]}>
    Update
  </button>
  <button onClick={[Function: onClick]}>
    Remove
  </button>
</div>"
`;
